### ISSUE TITLE
Move ShouldInstrumentAssemblyFile to InstrumentationContext

### ISSUE THREAD

**Issue by ffMathy** [lucaslorentz/minicover#136]
I am developing a CLI called Pruner (https://github.com/pruner/cli) that currently uses AltCover and GIT together to only run tests that were affected by your GIT changes. Both on the build server, but also when you jump in and see your colleague's branch. You only ever need to run tests that others didn't already run, or tests that are affected by your code changes.

It works, but right now requires that all my users install AltCover to their test projects. What I would much rather like to do, is not have that step at all, and instead have some sort of "Pruner" NuGet package instead, that the CLI uses, but where installing it into the test projects is not needed. This could be achieved by Minicover, which has the perfect level of abstraction, is easy to understand and read, and is very well engineered exactly to what I want to do.

So what my CLI basically needs, is to know which lines of code was run through by which tests. That could be achieved by #130.

However, right now, I would also like to be able to control which assemblies are being picked for instrumentation. That way, I can decide on my own if assemblies that have already been instrumented need to be re-instrumented, because their origin DLL changed.

In other words, if the incremental build of `dotnet build` only changed `Foo.dll` but not `Foo.Bar.dll`, only re-instrument `Foo.dll` because `Foo.Bar.dll` is already instrumented from last time.

If we moved `ShouldInstrumentAssemblyFile` to `InstrumentationContext` and made it `virtual`, that would solve all my problems I think, and then I can start contributing to Minicover, and also move everything over to it.

**Comment by ffMathy** [comment]
If this sounds like a good idea to you, I don't mind making a PR for you. I just wasn't sure it was the right way to go with it, and the project is already so well engineered that I didn't want to mess it up.

**Comment by lucaslorentz** [comment]
Very interesting project, I like it!

I would rather not move the content of that method because it checks things related to instrumentation, like additional files created and the need of PDB files to work.

Would it work if you did not put the assembly file in the instrumentation context at all?

We could create the additional method you suggested in the InstrumentationContext and call it, while keeping also the exiting method with its rules inside Instrumenter.

What do you think?
Feel free to create PRs

**Comment by ffMathy** [comment]
Ah of course - that makes sense 🤦‍♂️ 

I'll do that instead. Thank you!
